File: 53-maximum-subarray/main.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_sum = 0
        max_sum = float('-inf')
        n = len(nums)

        for i in range(n):
            cur_sum += nums[i]
            max_sum = max(max_sum, cur_sum)
            if cur_sum < 0:
                cur_sum = 0
        
        return max_sum

File: 53-maximum-subarray/main.go
func maxSubArray(nums []int) int {
    
    curSum, maxSum := 0, int(math.Inf(-1))
    n := len(nums)

    for i := range(n){
        curSum += nums[i]
        maxSum = max(maxSum, curSum)
        if curSum < 0 {
            curSum = 0
        }
    }

    return maxSum
}

func max(x, y int) int {
    if x < y { return y }
    return x
}
